src/GpsPoint.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Validator;
+
+final class GpsPoint extends AbstractValidator
+{
+
+    const OUT_OF_BOUNDS = 'gpsPointOutOfBounds';
+    const CONVERT_ERROR = 'gpsPointConvertError';
+    const INCOMPLETE_COORDINATE = 'gpsPointIncompleteCoordinate';
+
+    /**
+     * @var array
+     */
+    protected $messageTemplates = [
+        'gpsPointOutOfBounds' => '%value% is out of Bounds.',
+        'gpsPointConvertError' => '%value% can not converted into a Decimal Degree Value.',
+        'gpsPointIncompleteCoordinate' => '%value% did not provided a complete Coordinate',
+    ];
+
+    /**
+     * Returns true if and only if $value meets the validation requirements
+     *
+     * If $value fails validation, then this method returns false, and
+     * getMessages() will return an array of messages that explain why the
+     * validation failed.
+     *
+     * @param  mixed $value
+     * @return bool
+     * @throws Exception\RuntimeException If validation of $value is impossible
+     */
+    public function isValid($value)
+    {
+        if (strpos($value, ',') === false) {
+            $this->error(GpsPoint::INCOMPLETE_COORDINATE, $value);
+            return false;
+        }
+
+        list($lat, $long) = explode(',', $value);
```

Comment:
I made the change during merge, iirc.
On Jul 17, 2015 6:56 AM, "zf2timo" <notifications@github.com> wrote:

> In src/GpsPoint.php
> <https://github.com/zendframework/zend-validator/pull/18#discussion_r34882834>
> :
>
> > +     * If $value fails validation, then this method returns false, and
> > +     * getMessages() will return an array of messages that explain why the
> > +     * validation failed.
> > +     *
> > +     * @param  mixed $value
> > +     * @return bool
> > +     * @throws Exception\RuntimeException If validation of $value is impossible
> > +     */
> > +    public function isValid($value)
> > +    {
> > +        if (strpos($value, ',') === false) {
> > +            $this->error(GpsPoint::INCOMPLETE_COORDINATE, $value);
> > +            return false;
> > +        }
> > +
> > +        list($lat, $long) = explode(',', $value);
>
> @weierophinney <https://github.com/weierophinney> Yes, it's cleaner. But
> should i make this change, because it was already merged.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendframework/zend-validator/pull/18/files#r34882834>.
>